### PR TITLE
Pull out common UI tooling

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
* Added a `Improbable.Gdk.Core.Editor` asmdef.
* Moved the `SingletonScriptableObject<T>` there and made it public.
* Pulled out the `UiStateManager` from the `BuildConfigEditor` as I the same thing in the deployment launcher.

#### Tests
Tested build config, still worked. Rebased deployment launcher on these, still worked.

#### Documentation
Will need a changelog entry.

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/BuildConfigEditor.cs [hunk 1]
```diff
@@ -38,7 +39,7 @@ private class FoldoutState
         private BuildConfigEditorStyle style;
         private int invalidateCachedContent;
         private WorkerChoicePopup workerChooser;
-        private readonly Dictionary<int, object> stateObjects = new Dictionary<int, object>();
+        private UiStateManager stateManager = new UiStateManager();
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/BuildConfigEditor.cs]: `UiStateManager` or `UIStateManager`?

File: workers/unity/Packages/com.improbable.gdk.core/Editor/UiStateManager.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            if (!stateObjects.TryGetValue(hash, out var value))
+            {
+                value = new T();
+                stateObjects.Add(hash, value);
+            }
+
+            return (T) value;
+        }
+
+        public void SetStateObject<T>(int hash, T obj)
```
Comment (jessicafalk) [inline workers/unity/Packages/com.improbable.gdk.core/Editor/UiStateManager.cs]: this is currently not used.

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,6 +40,9 @@
 - Updated default snapshot to have more than one PlayerCreator entity.
 - Fixed package dependencies.
 - Worker flag changes are propagated to the `ViewDiff`.
+- Added a `Improbable.Gdk.Core.Editor` asmdef.
```
Comment (jared-improbable) [inline CHANGELOG.md]: asmdef looks like a keyboard-mashing typo :>

File: workers/unity/Packages/com.improbable.gdk.core/Editor/SingletonScriptableObject.cs [hunk 4]
```diff
@@ -4,9 +4,9 @@
 using UnityEditor;
 using UnityEngine;
 
-namespace Improbable.Gdk.BuildSystem.Configuration
+namespace Improbable.Gdk.Core.Editor
 {
-    internal abstract class SingletonScriptableObject<TSelf> : ScriptableObject
+    public abstract class SingletonScriptableObject<TSelf> : ScriptableObject
```
Comment (jared-improbable) [inline workers/unity/Packages/com.improbable.gdk.core/Editor/SingletonScriptableObject.cs]: Docstrings?

File: workers/unity/Packages/com.improbable.gdk.core/Editor/UiStateManager.cs (hunk 2 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Editor/UiStateManager.cs]: Its used in the fps PR 😉

File: CHANGELOG.md (hunk 3 above)
Comment (jamiebrynes7) [reply CHANGELOG.md]: Make sure to put this following line in your ashdfpafgd

File: workers/unity/Packages/com.improbable.gdk.core/Editor/SingletonScriptableObject.cs (hunk 4 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Editor/SingletonScriptableObject.cs]: Yes! Actually all of this needs docstrings

File: workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/BuildConfigEditor.cs (hunk 1 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/BuildConfigEditor.cs]: Sadly, the latter

**Pull request merged**